File: QID-3503-SFScola3/SFScola3.R
# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

x = read.table("Coca_cola.txt")     # Stock prices
x = ts(x)                           # transform to time series object

# Returns
r1 = (x[2:length(x)] - x[1:length(x) - 1])/x[1:length(x) - 1]   # Returns
r2 = log(x[2:length(x)]) - log(x[1:length(x) - 1])              # Log returns
split.screen(c(2, 1))

# Time series r1
screen(1)
plot(r1, type = "l", col = "blue", xlab = "Time (days)", ylab = "Returns")
# Time series r2
screen(2)
plot(r2, type = "l", col = "blue", xlab = "Time (days)", ylab = "Log Returns") 

File: QID-3503-SFScola3/SFScola3.m
% ---------------------------------------------------------------------
% Book:        SFS
% ---------------------------------------------------------------------
% Quantlet:    SFScola3
% ---------------------------------------------------------------------
% Description: SFScola3 reads daily stock prices for Coca-Cola company
%              from 2 January 2002 to 30 November 2004 and plots
%              the time series of returns (r1) and log returns (r2)
%              Corresponds to exercise 13.20(4.) in SFS
% ---------------------------------------------------------------------
% Usage:       -
% ---------------------------------------------------------------------
% Inputs:      none
% ---------------------------------------------------------------------
% Output:      Time series plots        
% ---------------------------------------------------------------------
% Example:     -
% ---------------------------------------------------------------------
% Author:      Julius Mungo, Andrija Mihoci  20110201
% ---------------------------------------------------------------------

clear all
close all
clc


x  = load('Coca_cola.txt');                             % Stock prices
r1 = diff(x)./x(2 : end, :);                           % Returns
r2 = diff(log(x));                                     % Log returns
subplot(2, 1, 1), plot(r1);                            % Time series r1
xlabel('Time (days)'); ylabel('Returns');              % x and y labels
subplot(2, 1, 2), plot(r2);                            % Time series r2
xlabel('Time (days)'); ylabel('Log returns');          % x and y labels


